lib/main.js

```diff
@@ -112,67 +162,76 @@ export default {
-
-          let { rulesDirectory } = configuration;
-          if (rulesDirectory) {
-            const configurationDir = path.dirname(configurationPath);
-            if (!Array.isArray(rulesDirectory)) {
-              rulesDirectory = [rulesDirectory];
-            }
-            rulesDirectory = rulesDirectory.map((dir) => {
-              if (path.isAbsolute(dir)) {
-                return dir;
-              }
-              return path.join(configurationDir, dir);
-            });
-
-            if (this.rulesDirectory) {
-              rulesDirectory.push(this.rulesDirectory);
-            }
-          }
+      lint: async (textEditor) => {
+        if (this.ignoreTypings && textEditor.getPath().toLowerCase().endsWith('.d.ts')) {
+          return [];
+        }
 
-          const linter = new Linter({
-            formatter: 'json',
-            rulesDirectory,
-          });
+        return this.lint(textEditor);
+      },
+    };
+  },
 
-          linter.lint(filePath, text, configuration);
-          const lintResult = linter.getResult();
+  lint(textEditor, options) {
+    const filePath = textEditor.getPath();
+    const text = textEditor.getText();
 
-          if (textEditor.getText() !== text) {
-            // Text has been modified since the lint was triggered, tell linter not to update
-            return null;
-          }
+    return this.getLinter(filePath).then((Linter) => {
+      const configurationPath = Linter.findConfigurationPath(null, filePath);
+      const configuration = Linter.loadConfigurationFromPath(configurationPath);
 
-          if (!lintResult.failureCount) {
-            return [];
+      let { rulesDirectory } = configuration;
+      if (rulesDirectory) {
+        const configurationDir = path.dirname(configurationPath);
+        if (!Array.isArray(rulesDirectory)) {
+          rulesDirectory = [rulesDirectory];
+        }
+        rulesDirectory = rulesDirectory.map((dir) => {
+          if (path.isAbsolute(dir)) {
+            return dir;
           }
-
-          return lintResult.failures.map((failure) => {
-            const startPosition = failure.getStartPosition().getLineAndCharacter();
-            const endPosition = failure.getEndPosition().getLineAndCharacter();
-            return {
-              type: 'Warning',
-              text: `${failure.getRuleName()} - ${failure.getFailure()}`,
-              filePath: path.normalize(failure.getFileName()),
-              range: [
-                [startPosition.line, startPosition.character],
-                [endPosition.line, endPosition.character],
-              ],
-            };
-          });
+          return path.join(configurationDir, dir);
         });
-      },
-    };
+
+        if (this.rulesDirectory) {
+          rulesDirectory.push(this.rulesDirectory);
+        }
+      }
+
+      const linter = new Linter(Object.assign({
+        formatter: 'json',
+        rulesDirectory,
+      }, options));
```

Comment:
Does it handle `undefined` being passed in here? If not, you need a default specified up on [L179](https://github.com/AtomLinter/linter-tslint/pull/149/files#diff-16b7d75b70953cbf4755508170f2f24cR179).